### PR TITLE
feat(new-buffer) : no new Buffer() error added

### PR DESCRIPTION
**Pull request by divyabiyani**
Changes w.r.t #1 .

Fixes https://github.com/mozilla/eslint-plugin-fxa/issues/1

### CODE REVIEW

File: lib/rules/no-new-buffer.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+module.exports = {
+  create(context) {
+    return {
+      NewExpression(node) {
+        const callee = node.callee
+        if (callee && callee.name === 'Buffer') {
+          context.report({
+            node: node,
+            message: 'new Buffer() is deprecated'
```
Comment (vladikoff) [inline lib/rules/no-new-buffer.js]: @divyabiyani let's include the code from: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/rules/no-new-buffer.js

It has suggestions to use `.from`, etc.

File: lib/configs/server.js [hunk 2]
```diff
@@ -11,6 +11,7 @@ module.exports = {
     'curly': 0,
     'semi': [2, 'never'],
 
-    'fxa/async-crypto-random': 'error'
+    'fxa/async-crypto-random': 'error',
+    'fxa/no-new-buffer': 'error'
```
Comment (vladikoff) [inline lib/configs/server.js:15]: 👍

File: lib/rules/no-new-buffer.js (hunk 1 above)
Comment (vladikoff) [reply lib/rules/no-new-buffer.js]: @divyabiyani we also need to create/include a test file. example: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/test/no-new-buffer.js

**Comment by vladikoff** [comment]
@divyabiyani this is almost ready to go, could you please also send a work in progress PR for the auth-server to see how many changes we have to make to pass this new validation?

File: lib/rules/no-new-buffer.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  return (args.length > 0 && typeof args[0].value === 'number') ? 'alloc' : 'from';
+};
+
+module.exports = {
+  create(context) {
+    return {
+      NewExpression(node) {
+        const callee = node.callee
+        const method = inferMethod(node.arguments)
+				const range = [
```
Comment (seanmonstar) [inline lib/rules/no-new-buffer.js]: The spacing here looks off.

Comment (divyabiyani) [reply lib/rules/no-new-buffer.js]: @seanmonstar rectified it.

File: test/no-new-buffer.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+const RuleTester = require('eslint').RuleTester
+const rule = require('../lib/rules/no-new-buffer')
+
+const ruleTester = new RuleTester();
+
+const allocError = {
+	message: '`new Buffer()` is deprecated, use `Buffer.alloc()` instead.'
```
Comment (vladikoff) [inline test/no-new-buffer.js]: @divyabiyani in this test file we probably want a `2-space` indent, see example: https://github.com/divyabiyani/eslint-plugin-fxa/blob/master/test/async-crypto-random.js

**Comment by vladikoff** [comment]
Fixes https://github.com/mozilla/eslint-plugin-fxa/issues/1

File: lib/configs/default.js [hunk 5]
```diff
@@ -47,7 +47,7 @@ module.exports = {
     'no-with': 2,
     'quotes': [2, 'single', 'avoid-escape'],
     'semi': [2, 'always'],
-    'space-unary-ops': 0,
+    'space-unary-ops': [2, {'nonwords': true , 'words': false, 'overrides': {'-': false, '+': false, '.': false}}],
```
Comment (vladikoff) [inline lib/configs/default.js:50]: @divyabiyani see https://github.com/mozilla/fxa-auth-server/pull/1624/files#r98958100

**Comment by vladikoff** [comment]
the buffer changes landed in https://github.com/mozilla/eslint-plugin-fxa/pull/3

**Pull request closed without merging**